ddtrace/tracer/option.go

```diff
@@ -556,3 +579,43 @@ func StackFrames(n, skip uint) FinishOption {
 		cfg.SkipStackFrames = skip
 	}
 }
+
+// parseHTTPCodeRanges parses HTTP codes range pairs and invalidates invalid formats.
+func parseHTTPCodeRanges(r string) []int {
+	codes := []int{}
+	for _, code := range strings.Split(r, ",") {
+		code = strings.TrimSpace(code)
+		if code == "" {
+			continue
+		}
+		valid, _ := regexp.MatchString("\\d{3}(?:-\\d{3})*(?:,\\d{3}(?:-\\d{3})*)*", code)
+		if !valid {
+			log.Warn("Invalid range for %v", code)
+			continue
+		}
+		rg := strings.Split(code, "-")
+		if len(rg) == 1 {
+			val, _ := strconv.Atoi(rg[0])
+			codes = insertCode(codes, val)
+		} else {
+			if rg[0] > rg[1] {
+				rg[0], rg[1] = rg[1], rg[0]
+			}
+			min, _ := strconv.Atoi(rg[0])
+			max, _ := strconv.Atoi(rg[1])
+			for i := min; i <= max; i++ {
+				codes = insertCode(codes, i)
+			}
+		}
+	}
+	return codes
+}
+
+// insertCode inserts the HTTP code to the list of codes in numerical order and ensures no duplicates.
+func insertCode(codes []int, c int) []int {
```

Comment:
We need to try and define functions as stand-alone outside the context in which they are used. If we look closer at this function, it does not insert an HTTP code. It just inserts an integer into a slice of integers and returns another slice of integers, so we should define it as such.

As an example, an appropriate definition here would be:
```suggestion
// appendOrdered appends n to the slice s at the necessary point in order to keep the result an ordered slice of integers.
// A new slice containing n is returned
func appendOrdered(s []int, n int) []int {
```
Do you see what I mean? To not define it in terms of where it is used, define it as standalone. Then it becomes universal.